### PR TITLE
In memory caching of HTML render and API requests

### PR DESCRIPTION
**Pull request by jonsherrard**
#### What have you done
I've added an in [memory cache](https://github.com/isaacs/node-lru-cache). It will cache the last 100 responses to the API and the renderHTML method, for 2 minutes each

#### Why have you done it
This drastically improves response times for the top 100 hit pages of a Tapestry site

#### Testing
The end-to-end integration tests still work, and there's not much code here so I feel like restructuring, code into unit testable methods (assert "loadFromCache" was called etc) would be testing node-lru-cache rather than tapestry.

### CODE REVIEW

File: src/server/handle-api.js [hunk 1]
```diff
@@ -1,13 +1,39 @@
...
+        fetch(remote)
+          .then(response => response.json())
+          .then(resp => {
+            const data = ('0' in resp) || resp instanceof Array ?
+              resp[0] :
+              resp
+            // We can only get here if there's nothing cached
+            // Put the response into the cache using the request path as a key
+            cache.set(remote, data)
+            console.log(`Loaded response from API: ${remote}`)
```
Comment (jonsherrard) [inline src/server/handle-api.js]: I can replace this with our own logging funcsh probs to get silent logging in tests... I think.

**Comment by thomasdigby** [comment]
Does this affect the loaders in their current state?

**Comment by jonsherrard** [comment]
Nope it's just the API response, so the Loaders don't "know" any thing has changed. Need a way to turn it off though because I was messing around with some other bits last night and got thoroughly confused by the cache.

Also the fact that nginx cache ignores WP-API query parameters...

**Comment by thomasdigby** [comment]
Awesome, is this good to merge then or do you want to add the ability to turn off the cached response?

**Comment by thomasdigby** [comment]
I've never been more ready to merge a PR. Is it good to go? @jonsherrard

**Comment by jonsherrard** [comment]
I'm nearly there @thomasdigby. I'm going to set magAge to 0, unless `process.env.NODE_ENV === 'production'`, it has already caused me numerous problems trying to develop with it on.

File: src/utilities/cache-manager.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import LRU from 'lru-cache'
+
+const internalCaches = []
+
+export default class CacheManager {
+
+  static createCache (name) {
+    internalCaches[name] = LRU({
+      max: 100,
+      maxAge: (process.env.NODE_ENV === 'production') ?
```
Comment (ecoad) [inline src/utilities/cache-manager.js]: What if you wanted to test caching is working on local or staging?

**Pull request merged**